### PR TITLE
Added options support to isAlphanumeric

### LINKED ISSUE #1026: Add options support to isAlphanumeric
isAlphanumeric is meant to support options in the same way that express-validator already supports them on isAlpha. I would like to add support for options on isAlphanumeric.

The validator docs list an identical description of `options` for both isAlpha and isAlphanumeric:
>options is an optional object that can be supplied with the following key(s): ignore which can either be a String or RegExp of characters to be ignored e.g. " -" will ignore spaces and -'s.

### PR DESCRIPTION
**Pull request by trumankain**
## Description

Added options support to isAlphanumeric, closes #1026 

## To-do list

- [ ] I have added tests for what I changed.
- [x] This pull request is ready to merge.

### CODE REVIEW

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks! LGTM with a couple of notes. Tests are not needed in this case because all the validators are supposed to be tested by `validatorjs`.

File: src/options.ts [hunk 1]
```diff
@@ -397,6 +397,10 @@ export interface IsAlphaOptions {
   ignore?: string | string[] | RegExp;
 }
 
+export interface IsAlphanumericOptions {
+  ignore?: string | string[] | RegExp;
```
[the remark this answers is not in the document]
Comment (trumankain) [reply src/options.ts:401]: Ah ok, sounds good!

File: src/chain/validators.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export interface Validators<Return> {
   equals(comparison: string): Return;
   isAfter(date?: string): Return;
   isAlpha(locale?: Options.AlphaLocale, options?: Options.IsAlphaOptions): Return;
-  isAlphanumeric(locale?: Options.AlphanumericLocale): Return;
+  isAlphanumeric(locale?: Options.AlphanumericLocale, options?: Options.IsAlphanumericOptions): Return;
```
[the remark this answers is not in the document]
Comment (trumankain) [reply src/chain/validators.ts:23]: Will do!

[the remark this answers is not in the document]
Comment (trumankain) [reply src/chain/validators.ts]: Ok, that file is now passing but looks like Travis still failing due to some warnings from other files? @fedeci

File: src/chain/validators-impl.ts [hunk 3]
```diff
@@ -93,8 +93,9 @@ export class ValidatorsImpl<Chain> implements Validators<Chain> {
     const ignore = Array.isArray(options?.ignore) ? options?.ignore.join('') : options?.ignore;
     return this.addStandardValidation(validator.isAlpha, locale, { ...options, ignore });
   }
-  isAlphanumeric(locale?: Options.AlphanumericLocale) {
-    return this.addStandardValidation(validator.isAlphanumeric, locale);
+  isAlphanumeric(locale?: Options.AlphanumericLocale, options?: Options.IsAlphanumericOptions) {
+    const ignore = options?.ignore;
+    return this.addStandardValidation(validator.isAlphanumeric, locale, { ...options, ignore });
```
Comment (fedeci) [inline src/chain/validators-impl.ts:98]:
```suggestion
    return this.addStandardValidation(validator.isAlphanumeric, locale, options);
```
Sorry for the nitpicks😄

**Review by fedeci — APPROVED** [review]
The failing CI is unrelated, if you could run prettier also on `CONTRIBUTING.md` it would be great, so that we fix the failing main branch.

File: src/chain/validators-impl.ts (hunk 3 above)
Comment (trumankain) [reply src/chain/validators-impl.ts]: no problem! I've tried running prettier on contributing multiple different ways but the file is not changing.

**Comment by fedeci** [comment]
Thanks!

**Comment by gustavohenke** [comment]
Published in v6.12.0!

**Pull request merged**